src/Bicep.Core/Registry/AzureContainerRegistryManager.cs

```diff
@@ -56,7 +56,7 @@ public async Task<OciArtifactResult> PullArtifactAsync(RootConfiguration configu
             return new OciArtifactResult(manifestDigest, manifest, manifestStream, moduleStream);
         }
 
-        public async Task PushArtifactAsync(Configuration.RootConfiguration configuration, OciArtifactModuleReference moduleReference, StreamDescriptor config, params StreamDescriptor[] layers)
+        public async Task PushArtifactAsync(RootConfiguration configuration, OciArtifactModuleReference moduleReference, string? artifactType, StreamDescriptor config, params StreamDescriptor[] layers)
```

Comment:
Should `artifactType` be non-nullable here if all modules published with Bicep >= 0.14 will include it?

Reply:
I needed to push an artifact without `artifactType` in one of the tests I added to ensure we can successfully pull modules pushed by previous versions of Bicep (which won't have `artifactType` set).